pkg/daemon/ceph/osd/volume_test.go

```diff
@@ -260,6 +268,97 @@ var cephVolumeRAWTestResult = `{
 }
 `
 
+func TestInitializeBlockPVCResult(t *testing.T) {
+	executor := &exectest.MockExecutor{}
+	executor.MockExecuteCommandWithCombinedOutput = func(command string, args ...string) (string, error) {
+		logger.Infof("%s %v", command, args)
+		if args[1] == "ceph-volume" && args[2] == "raw" && args[3] == "prepare" && args[4] == "--bluestore" && args[7] == "--crush-device-class" {
+			return initializeBlockPVCTestResult, nil
+		} else if args[1] == "ceph-volume" && args[2] == "lvm" && args[3] == "prepare" && args[4] == "--bluestore" && args[7] == "--crush-device-class" {
+			return initializeBlockPVCTestResult, nil
+		}
+
+		return "", errors.Errorf("unknown command %s %s", command, args)
+	}
+	os.Setenv(oposd.CrushDeviceClassVarName, "foo")
+	defer os.Setenv(oposd.CrushDeviceClassVarName, "")
+	context := &clusterd.Context{Executor: executor}
+	clusterInfo := &cephconfig.ClusterInfo{
+		CephVersion: cephver.CephVersion{Major: 14, Minor: 2, Extra: 8},
+	}
+	a := &OsdAgent{cluster: clusterInfo, nodeName: "node1"}
+
+	devices := &DeviceOsdMapping{
+		Entries: map[string]*DeviceOsdIDEntry{
+			"metadata": {Data: -1, Metadata: []int{0, 1}, Config: DesiredDevice{Name: "sda"}},
+		},
+	}
+
+	blockPath, metadataBlockPath, err := a.initializeBlockPVC(context, devices, false)
+	assert.Nil(t, err)
+	assert.Equal(t, "", blockPath)
+	assert.Equal(t, "", metadataBlockPath)
```

Comment:
Tests with `metadata` does not look right. With `metadata` we  should be asserting `metadataBlockPath` with some expected output rather than empty string.  [reference](https://github.com/rook/rook/blob/1ea73ce1233fcaa4a1b5518967eef2dcbd0cade6/pkg/daemon/ceph/osd/volume.go#L269)

Reply:
I changed the input and it's working but I'll verify the changes on the cluster.